authors/apps/articles/views.py

```diff
@@ -610,3 +613,69 @@ def get(self, request, slug):
             },
                 status=status.HTTP_404_NOT_FOUND
             )
+
+
+class GetAllArticleReports(generics.ListAPIView):
+    """ admin class for getting all article reports """
+    serializer_class = ReportSerializer
+    permission_classes = (IsAuthenticated,)
+    renderer_classes = (ReportJSONRenderer,)
+
+    def get(self, request):
+        """ get endpoint for fetching actual reports """
+        user = request.user
+        if user.is_staff is False:
+            return Response({
+                "error": "only admins may view all reports"
+            }, status.HTTP_401_UNAUTHORIZED)
+        all_reports = ReportArticle.objects.all()
+        serializer = self.serializer_class(all_reports, many=True)
+        return Response(serializer.data, status.HTTP_200_OK)
+
+
+class ReportAnArticle(APIView):
+    """
+    User class for reporting articles
+    a user may report olny one article ata time
+    """
+    serializer_class = ReportSerializer
+    permission_classes = (IsAuthenticated,)
+    renderer_classes = (ReportJSONRenderer,)
+
+    def post(self, request, slug):
+        """ post method for creating a single report instance """
+        data = request.data
+        user = request.user
+        message = data.get('message')
+        article = Article.objects.filter(
+            slug=slug, published=True, activated=True).first()
```

Comment:
@BrianSerem, this action can be transferred to the serializer class.

Reply:
Let me transfer this to the serializer class